File: src/MyServiceBus/Serialization/EnvelopeSerializerFactory.cs
namespace MyServiceBus.Serialization;

public sealed class EnvelopeSerializerFactory : ISerializerFactory
{
    private readonly IMessageHeaderConvention _headerConvention;

    public EnvelopeSerializerFactory()
        : this(MassTransitHeaderConvention.Instance)
    {
    }

    public EnvelopeSerializerFactory(IMessageHeaderConvention headerConvention)
    {
        _headerConvention = headerConvention;
    }

    public string ContentType => InboundMessageResolver.EnvelopeContentType;

    public IMessageSerializer CreateSerializer() => new EnvelopeMessageSerializer(_headerConvention);

    public IMessageDeserializer CreateDeserializer() => new EnvelopeMessageDeserializer(_headerConvention);
}

File: src/Java/myservicebus/src/main/java/com/myservicebus/serialization/EnvelopeSerializerFactory.java
package com.myservicebus.serialization;

public final class EnvelopeSerializerFactory implements SerializerFactory {
    private final MessageHeaderConvention headerConvention;

    public EnvelopeSerializerFactory() {
        this(MassTransitHeaderConvention.INSTANCE);
    }

    public EnvelopeSerializerFactory(MessageHeaderConvention headerConvention) {
        this.headerConvention = headerConvention;
    }

    @Override
    public String getContentType() {
        return DefaultInboundMessageResolver.ENVELOPE_CONTENT_TYPE;
    }

    @Override
    public MessageSerializer createSerializer() {
        return new EnvelopeMessageSerializer(headerConvention);
    }

    @Override
    public MessageDeserializer createDeserializer() {
        return new EnvelopeMessageDeserializer(headerConvention);
    }
}
